src/path.c

```diff
@@ -236,6 +236,146 @@ void twin_path_arc(twin_path_t *path,
     twin_path_set_matrix(path, save);
 }
 
+static twin_angle_t vector_angle(twin_fixed_t ux,
+                                 twin_fixed_t uy,
+                                 twin_fixed_t vx,
+                                 twin_fixed_t vy)
+{
+    twin_fixed_t dot = twin_fixed_mul(ux, vx) + twin_fixed_mul(uy, vy);
+
+    twin_fixed_t ua =
+        twin_fixed_sqrt(twin_fixed_mul(ux, ux) + twin_fixed_mul(uy, uy));
+    twin_fixed_t va =
+        twin_fixed_sqrt(twin_fixed_mul(vx, vx) + twin_fixed_mul(vy, vy));
+
+    /* cos(theta) = dot / (|u| * |v|) */
```

Comment:
It is not recommended to write `dot / (|u| * |v|)` directly, as `dot` is a custom variable name. It is suggested to rewrite it as `(u dot v) / (|u| * |v|)` and clarify that dot represents the dot product operator. Alternatively, you could describe it in words, stating that `cos(theta)` equals the dot product of `u` and `v` divided by the product of their magnitudes.